Спринт 6

disochkin opened the pull request:
Первая попытка

disochkin committed:
1. Убрал ограничение на длину списка для хранения истории просмотра задач.
2. Повторно просмотренные задачи не сохраняются в истории просмотра.
3. Обновил юнит-тесты.

disochkin committed:
1. Убрал ограничение на длину списка для хранения истории просмотра задач.
2. Повторно просмотренные задачи не сохраняются в истории просмотра.
3. Обновил юнит-тесты.

disochkin committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	src/controllers/InMemoryHistoryManager.java

disochkin committed:
Исправление code-style

disochkin committed:
Исправление code-style
вторая попытка

disochkin committed:
Исправление code-style
 попытка 3

disochkin committed:
Исправление code-style
 попытка 4

avfyodorov commented on src/controllers/HistoryManager.java:
```diff
@@ -2,10 +2,14 @@
+import java.util.List;
 
 public interface HistoryManager {
     void updateHistory(Task viewedTask);
```
Хорошо было бы провести небольшой рефакторинг и переименовать методы в соответствии с техническим заданием.

avfyodorov commented on test/controllers/HistoryManagerTest.java:
```diff
@@ -40,39 +40,140 @@ void setupClass() {
     @Test
     void getHistoryTest() {
```
В принципе, ко всем этим тестам можно было бы добавить аннотации, например:

@DisplayName("Добавить в историю")

Ну и т.д.
Необязательно. Просто имейте в виду, что есть такая аннотация.

avfyodorov commented on src/controllers/InMemoryHistoryManager.java:
```diff
@@ -3,28 +3,101 @@
+        else {
+            lastItem.next = newNode;
+            newNode.prev = lastItem;
+            lastItem = newNode;
```
Добавляется в любом случае, можно немного упростить. 
```java
private void linkLast(Task task) {  
    final Node node = new Node(task, last, null);  
    if (first == null) { //Список пустой-в начало  
       first = node;  
    } else {  //Список не пустой-прикрепляем в конец  
       last.next = node;  
    }  
    last = node; //Сдвинуть указатель в любом случае  
}
```

avfyodorov reviewed (commented):
Добрый день! 

Отметил несколько моментов, которые можно было бы поправить, но, думаю, что Вы справитесь с ними и вне ревью. 🙂
Работа принята. 
 Успехов в обучении.

disochkin committed:
исправил замечания:
1. добавил аннотации в тесты
2. переименовал метод updateHistory -> add + рефакторинг
